server/src/main/java/com/testsigma/service/TestStepService.java

```diff
@@ -133,7 +133,7 @@ private TestStep updateDetails(TestStep testStep) {
     }
 
     public TestStep update(TestStep testStep) throws TestsigmaException {
-        if (testStep.getConditionType()==TestStepConditionType.LOOP_WHILE &&testStep.getMaxIterations()>101){
+        if (testStep.getConditionType()==TestStepConditionType.LOOP_WHILE && testStep.getMaxIterations()!=null && testStep.getMaxIterations()>101){
```

Comment:
condition should be > 100, since allowed boundary is 1 - 100

Reply:
will change to 100